### PR TITLE
Model Page/Location Page Componentization

### PR DESCRIPTION
**Pull request by goldblatt**
Route /p3 for dev only. Making a new model page to iterate on + making a quick chartModule to hold the title + details + chart

### CODE REVIEW

File: src/components/Charts/ChartModule.style.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  @media (max-width: 600px) {
+    padding: 1.5rem 1rem;
+  }
+`;
+
+export const Wrapper = styled.div`
+  max-width: 900px;
+  margin: auto;
+
+  .highcharts-axis-labels {
```
Comment (pnavarrc) [inline src/components/Charts/ChartModule.style.js:32]: We will probably move the chart styles to its own `style` file, but we can leave it for now.

File: src/screens/ModelPage/ModelPage_Phase3.js [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+                </>
+              );
+            })}
+          </MainContentInnerBody>
+        </MainContentInner>
+      </MainContentWrapper>
+    );
+  };
+
+  const renderMapContent = () => {
```
Comment (pnavarrc) [inline src/screens/ModelPage/ModelPage_Phase3.js:144]: Consider exporting the render functions from `ModelPage` that are not changed and using them here to avoid duplication and to make sure that they stay consistent (until we change them).

File: src/screens/ModelPage/ModelPage_Phase3.js [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+                  View projections for{' '}
+                  <span onClick={() => goTo('/us/' + stateId.toLowerCase())}>
+                    {stateName}
+                  </span>
+                </div>
+              </StyledNoResults>
+            </StyledNoResultsWrapper>
+            {renderMapContent()}
+          </div>
+        ) : (
```
Comment (pnavarrc) [inline src/screens/ModelPage/ModelPage_Phase3.js:233]: Consider creating a component for the "no data" state (or create a render function). We can do that in a separate PR too.

File: src/App.js [hunk 4]
```diff
@@ -4,6 +4,7 @@ import { ThemeProvider, StylesProvider } from '@material-ui/core/styles';
 import { BrowserRouter, Route, Redirect, Switch } from 'react-router-dom';
 
 import ModelPage from 'screens/ModelPage/ModelPage';
+import ModelPage_Phase3 from 'screens/ModelPage/ModelPage_Phase3';
```
Comment (igorkofman) [inline src/App.js:7]: i'm confused, i thought at standup yesterday we just decided to do it live?

File: src/components/Charts/ChartModule.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            <Chart />
+            {children}
+          </Wrapper>
+        </ChartContainer>
+      </Panel>
+    </>
+  );
+};
+
+const Panel = ({ children, title }) => {
```
Comment (igorkofman) [inline src/components/Charts/ChartModule.js:33]: is there a point to this wrapper?

File: src/screens/ModelPage/ModelPage_Phase3.js [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+      title: '[DEV] Projections',
+    },
+    {
+      title: '[DEV] Case Growth Rate',
+    },
+    {
+      title: '[DEV] Testing',
+    },
+  ];
+  const renderMainContent = () => {
```
Comment (igorkofman) [inline src/screens/ModelPage/ModelPage_Phase3.js:120]: renderX is not a reacty approach to this.  these should just be React functional components

File: src/screens/ModelPage/ModelPage_Phase3.js [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+  };
+
+  return (
+    <Wrapper>
+      {metaTags}
+      <ContentWrapper>
+        {renderHeader()}
+        {projections && !projections.primary ? (
+          <div>
+            <StyledNoResultsWrapper>
```
Comment (igorkofman) [inline src/screens/ModelPage/ModelPage_Phase3.js:212]: this should be factored out to be it's own component as pablo mentions

File: src/screens/ModelPage/ModelPage_Phase3.js (hunk 2 above)
Comment (goldblatt) [reply src/screens/ModelPage/ModelPage_Phase3.js:144]: ah i think these are pretty nested within the react component itself so it would be hard to remove? but I agree with the principle!

File: src/components/Charts/ChartModule.style.js (hunk 1 above)
Comment (goldblatt) [reply src/components/Charts/ChartModule.style.js:32]: cook gonna ack this for right now because i moved everything everywhere but yes agree and will do in an upcoming PR

File: src/components/Charts/ChartModule.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+          </Wrapper>
+        </ChartContainer>
+        {/* TODO(add in descriotsions here) */}
+      </Panel>
+    </>
+  );
+};
+
+const Panel = ({ children }) => {
+  return <div style={{}}>{children}</div>;
```
Comment (igorkofman) [inline src/components/Charts/ChartModule.js:33]: can we get rid of Panel - i think it's a noop

File: src/screens/ModelPage/ModelPage_Phase3.js (hunk 2 above)
Comment (goldblatt) [reply src/screens/ModelPage/ModelPage_Phase3.js]: okay i was confused but this is now done!

File: src/screens/ModelPage/ModelPage_Phase3.js [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+          <LocationPageHeader projections={projections} />
+          <MainContentInnerBody>
+            {chart_modules.map((chart_module, index) => {
+              return (
+                <>
+                  {projections && (
+                    <ChartModule
+                      title={chart_module.title}
+                      series={['test']}
+                    ></ChartModule>
```
[the remark this answers is not in the document]
Comment (goldblatt) [reply src/screens/ModelPage/ModelPage_Phase3.js]: done

File: src/components/Charts/ChartModule.js (hunk 5 above)
Comment (goldblatt) [reply src/components/Charts/ChartModule.js]: removed

File: src/components/Charts/ModelChart.js [hunk 10]
```diff
@@ -40,14 +41,15 @@ const ModelChart = ({
...
   projections,
-  currentIntervention,
-  lastUpdatedDate,
   forCompareModels, // true when used by CompareInterventions.js component.
   stateId,
   selectedCounty,
 }) => {
   // We use the inferred projection if supported, otherwise the worst case for the currently active intervention
   let projection = projections.primary;
+  const currentIntervention = STATE_TO_INTERVENTION[stateId];
```
Comment (igorkofman) [inline src/components/Charts/ModelChart.js:50]: this is available as a field on projections ("stateIntevention") and should be exposed wit ha getter

File: src/components/ModelView/ChartsHolder.tsx [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+  MainContentInner,
+  MainContentInnerBody,
+} from './ChartsHolder.style';
+import LocationPageHeader from 'components/LocationPageHeader/LocationPageHeader';
+import NoCountyDetail from './NoCountyDetail';
+import ModelChart from 'components/Charts/ModelChart';
+import { Projections } from 'models/Projections';
+import { Projection } from 'models/Projection';
+
+const ChartsHolder = (props: {
```
Comment (igorkofman) [inline src/components/ModelView/ChartsHolder.tsx:14]: Why do you prefer to do it this way vs ({a:string,b:string..})

File: src/components/ModelView/ChartsHolder.tsx [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+  projections: Projections; // make this be a type with
```
Comment (igorkofman) [inline src/components/ModelView/ChartsHolder.tsx:15]: cut-off comment?

File: src/components/ModelView/ChartsHolder.tsx [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+  stateId: string;
+  countyId: string;
+}) => {
+  // this is the inferred projection. Ideally we'd pass in just the single projeciton, but the location page header needs the full obj
```
Comment (igorkofman) [inline src/components/ModelView/ChartsHolder.tsx:19]: technically "primary" might not be inferred if we don't have an inferred one for this county.

File: src/components/ModelView/ChartsHolder.tsx [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+  const projection: Projection = props.projections.primary;
+
+  if (projection && projection.isInferred) {
+    const rt_data = projection.getDataset('rt');
```
Comment (igorkofman) [inline src/components/ModelView/ChartsHolder.tsx:23]: camelCase

File: src/screens/ModelPage/ModelPage.js [hunk 15]
```diff
@@ -92,206 +61,42 @@ function ModelPage() {
...
-                  </span>
-                </div>
-              </StyledNoResults>
-            </StyledNoResultsWrapper>
-            {renderMapContent()}
-          </div>
-        ) : (
-          [renderMainContent(), renderMapContent()]
-        )}
+        <SearchHeaderWrapper>
```
Comment (igorkofman) [inline src/screens/ModelPage/ModelPage.js:75]: We do a lot of these wrappers that seem like they should just be implementation details of the underlying component..   Maybe I'm missing something, but this should just be <SearchHeader ../>, no?

File: src/screens/ModelPage/ModelPage.js [hunk 16]
```diff
@@ -92,206 +61,42 @@ function ModelPage() {
...
+          <SearchHeader
+            setMapOption={setMapOption}
+            mobileMenuOpen={mobileMenuOpen}
+            setMobileMenuOpen={setMobileMenuOpen}
+          />
+        </SearchHeaderWrapper>
+        <ChartsHolder
```
Comment (igorkofman) [inline src/screens/ModelPage/ModelPage.js:82]: reading this I'm not totally sure what a "chartsholder" is.  i guess it holds some charts?  It's weird that it contains the page header in it since that has nothing to do with charts..

also i'd consider renaming ModelPage to "LocationPage"

File: src/components/ModelView/ChartsHolder.tsx (hunk 11 above)
Comment (goldblatt) [reply src/components/ModelView/ChartsHolder.tsx:14]: I don't think you can do that in typescript-- i'd have to do something like 

```
const Component = ({
  a,
  b,
}: {
  a: string;
  b: string;
}) => {...}
```

or hmmmm i guess let me mess with the generic types

Comment (goldblatt) [reply src/components/ModelView/ChartsHolder.tsx:14]: I think the FunctionComponent expects props: Type

File: src/screens/ModelPage/ModelPage.js (hunk 15 above)
Comment (goldblatt) [reply src/screens/ModelPage/ModelPage.js:75]: ya i mean ideally but i'd rather not mess too much with the css (this pr is already a lot and i'm just trying to move towards a lil more components?

File: src/screens/ModelPage/ModelPage.js (hunk 16 above)
Comment (goldblatt) [reply src/screens/ModelPage/ModelPage.js:82]: cool  ya this seems out of scope but i changed to locationpage

Comment (goldblatt) [reply src/screens/ModelPage/ModelPage.js:74]: i got a lil nervus and left it mostly the same since i was changing so many other things. made https://github.com/covid-projections/covid-projections/issues/598 to address in the future
